src/store.rs

```diff
@@ -1,68 +1,43 @@
 use rayon::prelude::*;
-use std::collections::VecDeque;
```

Comment:
This is still a huge file, and still contains a crapton of commented lines that show this is in flux. Could we:
- clean this file up?
- use macros to reduce ithe repeated pattern matches that occur everywhere?